### ISSUE TITLE
How can I get the untranslated text in Braille Display Driver

### ISSUE THREAD

**Issue by tibabit** [nvaccess/nvda#7151]
Hi I am writing a brailleDisplayDriver but I do not want to use the translation provided by the NVDA, instead I want to get the ASCII text and translate it whenever I require.

Is there any way to do it in the brailleDisplayDriver.

If there is any other way to access the text even without the braille drivers then please suggest.

**Comment by jcsteh** [comment]
Can you give full details concerning why you want to do this? There is currently no simple API for this. Even if you could get the translated text, scrolling, cursor display, cursor routing and selection would all be quite broken if you used your own translation instead of NVDA's, since NVDA must map all of these things to and from braille cells.

**Comment by tibabit** [comment]
Since NVDA does not handle 6 dot inputs, we are implementing it on our braille display. So we were hoping if we can communicate with NVDA using just ASCII not translated version. Also I see that these things(routing, cursor, window shifting) are implemented in braille.Range class, soI was hoping if it is possible to change it's behavior from the braille display driver itself, without modifying the core modules.

**Comment by jcsteh** [comment]
Contracted input is coming and is almost done; see #2439 and #6449. I'd definitely recommend waiting for this rather than implementing it in a driver, as NVDA can do this in a far more reliable, robust and consistent manner in the core.

Much of the braille core is centred around the concept of passing raw cells to a display and mapping to and from raw cells. Even if you managed to replace all of this, you'd need to reimplement a massive amount of functionality.

I'm closing this as won't fix, as this just isn't feasible and the use case isn't valid once contracted braille input lands.
